### ISSUE TITLE
Error caused by emoji in comments in yolov5/data/hyps/*.yaml file

### ISSUE THREAD

**Issue by sina-Kim** [ultralytics/yolov5#5040]
Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following,
otherwise it is non-actionable, and we can not help you:

- **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
- **Common dataset**: coco.yaml or coco128.yaml
- **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#environments

If this is a custom dataset/training question you **must include** your `train*.jpg`, `val*.jpg` and `results.png`
figures, or we can not help you. You can generate these with `utils.plot_results()`.

## 🐛 Bug

Decode error occurs when executing the command suggested for input after git clone in Windows environment

## To Reproduce (REQUIRED)

Input:

```
(env38) PS C:\Users\Username\PycharmProjects\yolov5> python train.py --img 640 --batch 16 --epochs 3 --data coco128.yaml --weights yolov5s.pt
```

Output:

```
Downloading https://ultralytics.com/assets/Arial.ttf to C:\Users\Username\AppData\Roaming\Ultralytics\Arial.ttf...
train: weights=yolov5s.pt, cfg=, data=coco128.yaml, hyp=data\hyps\hyp.scratch.yaml, epochs=3, batch_size=16, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, adam=False, sync_bn=False, workers=8, entity=None, project=runs\train, name=exp, exist_ok=False, quad=False, linear_lr=False, label_smoothing=0.0, upload_dataset=False, bbox_interval=-1, save_period=-1, artifact_alias=latest, local_rank=-1, freeze=0, patience=100
github: up to date with https://github.com/ultralytics/yolov5
YOLOv5  v5.0-493-g1922dde torch 1.8.0+cu111 CUDA:0 (GeForce RTX 3090, 24576.0MB)

Traceback (most recent call last):
  File "train.py", line 615, in <module>
    main(opt)
  File "train.py", line 512, in main
    train(opt.hyp, opt, device, callbacks)
  File "train.py", line 76, in train
    hyp = yaml.safe_load(f)  # load hyps dict
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\__init__.py", line 112, in load
    loader = Loader(stream)
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\loader.py", line 34, in __init__
    Reader.__init__(self, stream)
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\reader.py", line 85, in __init__
    self.determine_encoding()
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\reader.py", line 124, in determine_encoding
    self.update_raw()
  File "C:\Users\Username\miniconda3\envs\env38\lib\site-packages\yaml\reader.py", line 178, in update_raw
    data = self.stream.read(size)
UnicodeDecodeError: 'cp949' codec can't decode byte 0xf0 in position 9: illegal multibyte sequence
```

## Environment

- OS: Windows 10
- GPU : RTX3090

## Additional context

This error occurs because of the rocket-shaped emoji (🚀) in the yolov5/data/hyps/*.yaml file. You can fix the error by editing the yaml file or specifying the decoding method in detail.

**Comment by sina-Kim** [comment]
I used a translator. I'm sorry if the post is weird or the expression is rude.

**Comment by glenn-jocher** [comment]
@sina-Kim it looks like the best solution is to specify the decoding method. I think we do this is other loaders, not sure why it's not done here.
https://stackoverflow.com/questions/62844140/how-to-fix-unicodedecodeerror-cp949-codec-cant-decode-byte-0xeb-in-position

**Comment by glenn-jocher** [comment]
@sina-Kim can you try to replace train.py L76 with this?
```python
    # Hyperparameters
    if isinstance(hyp, str):
        with open(hyp, encoding='utf-8') as f:
            hyp = yaml.safe_load(f)  # load hyps dict
```

or 
```python
    # Hyperparameters
    if isinstance(hyp, str):
        with open(hyp, errors='ignore') as f:
            hyp = yaml.safe_load(f)  # load hyps dict
```

**Comment by sina-Kim** [comment]
@glenn-jocher it works!

**Comment by glenn-jocher** [comment]
@sina-Kim great, but exactly which solution works?

We need to implement this in master to help future users, please let us know which to implement. Thank you!

**Comment by sina-Kim** [comment]
@glenn-jocher The first example worked. (Actually, I haven't even tested the second one...)
```python
    # Hyperparameters
    if isinstance(hyp, str):
        with open(hyp, encoding='utf-8') as f:
            hyp = yaml.safe_load(f)  # load hyps dict
```

**Comment by glenn-jocher** [comment]
@sina-Kim good news 😃! Your original issue may now be fixed ✅ in PR #5060. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
